File: src/test/java/com/github/kayjamlang/tests/expressions/AccessExpressionNamedExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.Container;
import com.github.kayjamlang.core.expressions.NamedExpression;
import com.github.kayjamlang.core.expressions.AccessExpression;
import com.github.kayjamlang.core.expressions.VariableLinkExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertSame;

public class AccessExpressionNamedExpressionTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("test.test {}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(AccessExpression.class, expression.getClass());

        AccessExpression accessExpression = (AccessExpression) expression;
        assertSame(VariableLinkExpression.class, accessExpression.root.getClass());
        assertSame(NamedExpression.class, accessExpression.child.getClass());

        NamedExpression namedContainer = (NamedExpression) accessExpression.child;
        assertSame(Container.class, namedContainer.stmt.getClass());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/AccessExpressionNamedExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.Container
import com.github.kayjamlang.core.expressions.{AccessExpression, NamedExpression, VariableLinkExpression}
import org.junit.Assert.{assertNotNull, assertSame}
import org.junit.{BeforeClass, Test}


object AccessExpressionNamedExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("test.test {}"))
  }
}

class AccessExpressionNamedExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = AccessExpressionNamedExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[AccessExpression], expression getClass)
    val accessExpression = expression.asInstanceOf[AccessExpression]
    assertSame(classOf[VariableLinkExpression], accessExpression.root getClass)
    assertSame(classOf[NamedExpression], accessExpression.child getClass)
    val namedContainer = accessExpression.child.asInstanceOf[NamedExpression]
    assertSame(classOf[Container], namedContainer.expression getClass)
  }
}